File: p283e/move_zeroes.js
/**
 * @param {number[]} nums
 * @return {void} Do not return anything, modify nums in-place instead.
 */
var moveZeroes = function(nums) {
  let j = 0;
  for (let i = 0; i < nums.length; ++i) {
    if (nums[i] !== 0) nums[j++] = nums[i];
  }
  while (j < nums.length) nums[j++] = 0;
};
// TEST
[[], [1, 2], [0, 1], [2, 0, 1], [0, 1, 0, 3, 12]].forEach(function(test) {
  console.log('Move', test, 'zeros ->');
  moveZeroes(test);
  console.log(test);
});

File: p283e/move_zeroes.py
from typing import List


class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        j = 0
        for i in range(len(nums)):
            if nums[i] != 0:
                nums[j] = nums[i]
                j = j + 1
        while j < len(nums):
            nums[j] = 0
            j = j + 1


# TESTS
tests = [[], [1, 2], [0, 1], [2, 0, 1], [0, 1, 0, 3, 12]]
for t in tests:
    sol = Solution()
    sol.moveZeroes(t)
    print("Move zeros ->", t)
